libs/utils/src/lib/services/broadcast-channel/broadcast-channel.service.ts

```diff
@@ -0,0 +1,95 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { EMPTY, fromEvent, Observable } from 'rxjs';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class BroadcastChannelService {
+	private broadcastChannel: Record<string, BroadcastChannel> = {};
+
+	constructor(@Inject(PLATFORM_ID) private platformId: string) {}
+
+	/**
+	 * initChannel
+	 *
+	 * The initChannel method initializes a new BroadcastChannel instance.
+	 *
+	 * @param channelName{string} - The name of the Broadcast Channel.
+	 */
+	public initChannel(channelName: string): void {
+		if (!channelName) {
```

Comment:
This check is done here everywhere, but besides this single instance, is it really necessary?

Reply:
I've repeated the check for every method to avoid the code from breaking the host application since this will presumably be used for a non-essential features. The developer will be warned of something going wrong, but the flow of the application will not break.

I'll see if I can simplify the checks and streamline the messages and log them as errors to give them a higher priority within the flow.